### ISSUE TITLE
allow removing `name="date"` attribute from `date-picker` input

### ISSUE THREAD

**Issue by argshook** [schedule-x/schedule-x#839]
currently, the HTML of a rendered calendar includes `<input name="date"/>` (you can find it by searching for `data-testid="date-picker-input"`).

the input is from datepicker shown in the calendar header. It's meant to
be used to control the selected date of the calendar. However, because
of the `name` attribute, it becomes part of the form data.

Thus, if schedule-x calendar is rendered within a form, like so:

```tsx
const calendar = useCalendarApp(config)

// ...
<form>
  <input type="text" name="name" />
  <ScheduleXCalendar calendar={calendar} />
</form>
```

then the FormData (i.e. what gets submitted) would include 2 properties:
* `name` - explicit input set by user
* `date` - implicit input coming from calendar. Was a surprise for me!

things can get worse if user has a dynamic form where multiple calendars are shown.
then each would include a `date` and FormData would end up with multiple properties named with same `date` (it's allowed to have multiple entries with same name in FormData. which just adds to the confusion :D ).

To avoid this i'd like to disable `name` attribute completely. In a way this is like saying "thanks, schedulex, but i'd like to control my form data myself".

However, currently i can't remove the `name` attribute as it has a fallback value:

https://github.com/schedule-x/schedule-x/blob/35e20a17819781ca5206fe5986e7fc169088821a/packages/date-picker/src/components/app-input.tsx#L98 

it might make sense here to use `??` instead of `||` to allow for `null`.

for the user, a way to control this name, i thought, was via `datePicker` config, like this:

```ts
useCalendarApp({
  datePicker: {
    name: "my-date", // type for this field is defined, so this type-checks fine
  },
});
```

but it doesn't work for me. What worked was using `calendarControls` plugin:

```ts
useCalendarApp({
  plugins: [calendarControls()],
  callbacks: {
    beforeRender: () => {
      calendarControls.$app.datePickerConfig.name = "my-date";
    },
  },
});
```

but this feels awkward: i need to add extra callback and looks like i'm digging into some internal value.
ideally i'd like to not have the `name` attribute completely.

So, i could go an make a PR which removes this attribute. But as
schedule-x is already used in the wild, some users might be relying on
this data and removal would be a breaking change. So perhaps a better
improvement would be to make `datePicker.name` config work as expected
and allow null value?

wdyt?

**Comment by tomosterlund** [comment]
Yes, removing "name" completely would certainly break things, but I guess we could adjust the type to `string | null` and then remove the attribute on `null` like you say!

**Comment by tomosterlund** [comment]
will be open to PRs for this anytime, but I don't think I will find the time to implement this myself
